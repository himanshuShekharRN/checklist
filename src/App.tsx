import React from 'react';
import 'react-native-gesture-handler';

import {Navigation} from './navigation';
import {Provider} from 'react-redux';
import {persistor, store} from './store';
import {PersistGate} from 'redux-persist/integration/react';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};
