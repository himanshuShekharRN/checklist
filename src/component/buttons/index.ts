import {FloatingAddButton} from './floatingAddButton';
import {RoundedButton} from './roundedButton';

export {FloatingAddButton, RoundedButton};
