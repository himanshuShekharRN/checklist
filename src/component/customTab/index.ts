import {CustomTab} from './CustomTab';
import {FirstRoute} from './firstRoute';
import {SecondRoute} from './secondRoute';
import {ThirdRoute} from './thirdRoute';
import {FourthRoute} from './fourthRoute';
import {FifthRoute} from './fifthRoute';

export {
  CustomTab,
  FirstRoute,
  SecondRoute,
  ThirdRoute,
  FourthRoute,
  FifthRoute,
};
