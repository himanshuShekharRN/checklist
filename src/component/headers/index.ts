import {HeadersWithButton} from './headersWithButton/HeadersWithButton';

export {HeadersWithButton};
