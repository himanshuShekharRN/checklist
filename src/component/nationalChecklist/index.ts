import {NationalChecklist} from './NationalChecklist';

export {NationalChecklist};
