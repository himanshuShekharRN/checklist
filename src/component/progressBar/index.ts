import {LinearProgressBar} from './linearProgressBar';

export {LinearProgressBar};
