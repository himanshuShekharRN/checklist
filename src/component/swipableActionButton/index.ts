import {SwipableActionButton} from './SwipableActionButton';

export {SwipableActionButton};
