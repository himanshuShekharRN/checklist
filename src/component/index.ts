import {SwipableCards} from './swipableCards/SwipableCards';
import {SwipableActionButton} from './swipableActionButton/SwipableActionButton';
import {Card} from './cards';
import {MyCheckList} from './myChecklist';
import {Space} from './space';
import {PreDepartureList} from './preDepartureList';
import {Title} from './title';
import {HeadersWithButton} from './headers';
import {CustomSVG} from './customSVG';
import {CustomIcon} from './customIcon';
import {FloatingAddButton, RoundedButton} from './buttons';
import {ListCard} from './listCard';
import {LinearProgressBar} from './progressBar';
import {CustomTab} from './customTab';
import {DepartureDocCard} from './departureDocCard';
import {NationalChecklist} from './nationalChecklist';

export {
  FloatingAddButton,
  Card,
  MyCheckList,
  Space,
  PreDepartureList,
  Title,
  HeadersWithButton,
  CustomSVG,
  CustomIcon,
  RoundedButton,
  ListCard,
  LinearProgressBar,
  CustomTab,
  DepartureDocCard,
  NationalChecklist,
  SwipableActionButton,
  SwipableCards,
};
