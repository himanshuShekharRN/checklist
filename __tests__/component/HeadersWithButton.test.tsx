import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react-native';
import {HeadersWithButton} from '../../src/component';
import {Text} from 'react-native';

describe('HeadersWithButton', () => {
  const onPressBackIconHandlerMock = jest.fn();
  const containerStyle = {
    flex: 1,
  };
  const props = {
    onPressBackIconHandler: onPressBackIconHandlerMock,
    containerStyle: containerStyle,
    iconContainerStyle: containerStyle,
    testID: 'test',
  };

  const tree: any = render(
    <HeadersWithButton {...props}>
      <Text>Button Content</Text>
    </HeadersWithButton>,
  );

  test('renders correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
  });

  test('renders children correctly', () => {
    render(
      <HeadersWithButton {...props}>
        <Text>Button Content</Text>
      </HeadersWithButton>,
    );

    const buttonContent = screen.getByText('Button Content');
    expect(buttonContent).toBeTruthy();
  });

  test('calls onPressBackIconHandler when the header is pressed', () => {
    render(
      <HeadersWithButton {...props}>
        <Text>Button Content</Text>
      </HeadersWithButton>,
    );

    const iconElement = screen.getByTestId('test');
    fireEvent.press(iconElement);

    expect(onPressBackIconHandlerMock).toHaveBeenCalled();
  });

  test('applies custom iconContainerStyle', () => {
    render(
      <HeadersWithButton {...props}>
        <Text>Card Content</Text>
      </HeadersWithButton>,
    );

    const card = screen.getByTestId('test');
    expect(card.props.style).toContainEqual(containerStyle);
  });

  test('applies custom containerStyle', () => {
    render(
      <HeadersWithButton {...props}>
        <Text>Card Content</Text>
      </HeadersWithButton>,
    );

    const card = screen.getByTestId('headersWithButton');
    expect(card.props.style).toContainEqual(containerStyle);
  });

  test('snapshot created successfully', () => {
    render(
      <HeadersWithButton {...props}>
        <Text>Card Content</Text>
      </HeadersWithButton>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
