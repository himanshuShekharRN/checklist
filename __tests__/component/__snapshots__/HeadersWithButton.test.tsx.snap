// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadersWithButton snapshot created successfully 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      {
        "flex": 1,
      },
    ]
  }
  testID="headersWithButton"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 40,
        },
        {
          "flex": 1,
        },
      ]
    }
    testID="test"
  >
    <View
      testID="customIcon"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#000000",
              "fontSize": 40,
            },
            undefined,
            {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î—‹
      </Text>
    </View>
  </View>
  <Text>
    Card Content
  </Text>
</View>
`;
